ERCS/erc-7524.md

```diff
@@ -0,0 +1,211 @@
+2. Signers can use existing secpk256k1 keypairs, such as those in hardware wallets that support Ethereum accounts. As a consequence, secret keys can remain in secure enclaves if there is a generator point multiplication API into the enclave (which Ledger for instance has).
+
+### Parameters
+
+This scheme uses the secp256k1 curve, defined in SEC 2: Recommended Elliptic Curve Domain Parameters v2, page 9.
+
+We use the following notation to refer to the parameters of this curve:
+
+- $g$: the base point (also called the generator) of the curve.
+- $p$: the order of the curve.
+- $F_p$: the finite field whose order is $p$.
+
+Note we use exponential notation to denote elliptic curve scalar multiplications.
+
+### Public key encoding functions
+
+### SEC1
+
+This scheme uses the SEC1 elliptic curve point encoding scheme defined in SEC 1: Elliptic Curve Cryptography v2 . Point compression is used. We use the notation $\mathsf{sec1}(pk)$ to denote the compressed encoding of secp256k1 curve point $pk$ as a bytestring of length 33.
+
+### Hash functions
+
+**SHA256**
+
+This scheme uses the SHA256 hash function defined in IETF RFC4634.
+
+In this document, we use the notation $\mathsf{sha256}(a_1,.. a_n)$ to denote the sha256 digest of the concatentation of $n$ values $a_1, ..., a_n$. The digest should then be interpreted as a big-endian value in the secp256k1 scalar field.
+
+### Hash-to-curve
+
+We use the notation $\mathsf{htc}([a_1, ..., a_n])$ to denote the elliptic curve point which is the result of the `secp256k1_XMD:SHA-256_SSWU_RO_ (IETF Draft draft-irtf-cfrg-hash-to-curve-16 Appendix J.8.1)` hash-to-curve algorithm over the concatentation of $n$ values $a_1, ..., a_n$.
+
+### Key generation
+
+A *keypair* comprises of $(sk, pk)$, defined as such:
+
+- $sk$: The user's secret key, which is a cryptographically secure random scalar in the field $F_p$.
+- $pk$: The user's public key, defined as $g^{sk}$, which is a point on the secp256k1 curve.
+
+### Signature generation
+
+This scheme builds upon the Chaum-Pedersen signature scheme. Given a 32-byte message $m$ and a keypair $(sk, pk)$, a  user may generate a signature as such:
+
+1. Pick a random $r$ from $F_p$.
+2. Compute $h = \mathsf{htc}([m, \mathsf{sec1}(pk)])$.
+3. Compute $z = h ^ r$.
+4. Compute the nullifier $\mathsf{nul} = h^{sk}$.
+5. Compute $c = \mathsf{sha256}([g, pk, h, \mathsf{nul}, g^r, z]])$.
+6. Compute $s = r + sk \cdot c$.
+
+The signature is $(z, s, g^r, c, \mathsf{nul})$.
+
+The length of the input to $\mathsf{htc}$ is always 65 bytes.
+
+Note that in this scheme, we compute $h$ as the hash of the message and $pk$, not the message and $r$. This is to make our scheme deterministic.
+
+### Signature verification (non-ZK)
+
+********************Note: Non-ZK signature verification is not part of the proposal but relevant for an intuitive understanding of the ZK signature verification.********************
```

Comment:
Normally I'd say that if something isn't part of the specification, it doesn't belong in the specification section. In this case, I don't have enough domain knowledge to say whether this is sufficiently relevant or not. Feel free to leave it in.

I would, however, recommend against using long strings of asterisks and to use something like this instead:

```suggestion
> 📝 **Note:** This section is non-normative.
>
> Non-ZK signature verification is not part of this proposal but relevant for an intuitive understanding of the ZK signature verification.
```